lib/grppcatr.gi

```diff
@@ -480,18 +480,31 @@ end );
 ##
 #M  MaximalNormalSubgroups( <G> )
 ##
-InstallMethod( MaximalNormalSubgroups, "for solvable groups",
+InstallMethod( MaximalNormalSubgroups, "for abelian groups",
                [ IsGroup and IsAbelian ],
+               # IsGroup and IsFinite ranks higher than
+               # IsGroup and IsAbelian, so we have to increase the
+               # rank, otherwise the method for normal subgroup computation
+               # is selected.
+               RankFilter( IsGroup and IsFinite and IsAbelian )
+               - RankFilter( IsGroup and IsAbelian ),
```

Comment:
ok (unimportant nitpick: perhaps rewrap the comment, it should fit in three instead of four lines?)